### ISSUE TITLE
client: `max_buffer_capacity_per_subscription` doesn't check that n == 0

### ISSUE THREAD

**Issue by YichiZhang0613** [paritytech/jsonrpsee#1357]
In jsonrpsee/core/src/client/async_client/mod.rs, the comment indicates that there is a constraint max != 0 while the code doesn't check it. I think the code should use assert!(max > 0) to check it like functions above.
```rust
/// # Panics
	///
	/// This function panics if `max` is 0.
	pub fn max_buffer_capacity_per_subscription(mut self, max: usize) -> Self {
		self.max_buffer_capacity_per_subscription = max;
		self
	}
```

**Comment by niklasad1** [comment]
Sure, that would be good, are you down to create a PR for? 

Since it is just wrapper of over `toko::mpsc::channel` which has such assert but that occurs only if a subscription is created so would be better to panic in the builder.

**Comment by niklasad1** [comment]
Closed by #1358
